what4/src/What4/Solver/Yices.hs

```diff
@@ -434,36 +448,36 @@ instance SMTWriter (Connection s) where
   arrayUpdate a i v =
     T $ app "update" [ renderTerm a, builder_list (renderTerm <$> i), renderTerm v ]
 
-  commentCommand _ b = Cmd (";; " <> b)
+  commentCommand _ b = safeCmd (";; " <> b)
 
-  pushCommand _   = Cmd "(push)"
-  popCommand _    = Cmd "(pop)"
-  resetCommand _  = Cmd "(reset)"
-  checkCommand _  = Cmd "(check)"
+  pushCommand _   = safeCmd "(push)"
+  popCommand _    = safeCmd "(pop)"
+  resetCommand _  = safeCmd "(reset)"
+  checkCommand _  = safeCmd "(check)"
   checkWithAssumptionsCommand _ nms =
-    Cmd $ app_list "check-assuming" (map Builder.fromText nms)
+    safeCmd $ app_list "check-assuming" (map Builder.fromText nms)
 
-  getUnsatAssumptionsCommand _ = Cmd "(show-unsat-assumptions)"
-  getUnsatCoreCommand _ = Cmd "(show-unsat-core)"
+  getUnsatAssumptionsCommand _ = safeCmd "(show-unsat-assumptions)"
+  getUnsatCoreCommand _ = safeCmd "(show-unsat-core)"
   setOptCommand _ x o = setParamCommand x (Builder.fromText o)
 
-  assertCommand _ (T nm) = Cmd $ app "assert" [nm]
-  assertNamedCommand _ (T tm) nm = Cmd $ app "assert" [tm, Builder.fromText nm]
+  assertCommand _ (T nm) = unsafeCmd $ app "assert" [nm]
+  assertNamedCommand _ (T tm) nm = unsafeCmd $ app "assert" [tm, Builder.fromText nm]
 
   declareCommand _ v args rtp =
-    Cmd $ app "define" [Builder.fromText v <> "::"
-                        <> unType (fnType (toListFC yicesType args) (yicesType rtp))
-                       ]
+    safeCmd $ app "define" [Builder.fromText v <> "::"
+                            <> unType (fnType (toListFC yicesType args) (yicesType rtp))
+                           ]
 
   defineCommand _ v args rtp t =
-    Cmd $ app "define" [Builder.fromText v <> "::"
-                         <> unType (fnType ((\(_,tp) -> viewSome yicesType tp) <$> args) (yicesType rtp))
-                       , renderTerm (yicesLambda args t)
-                       ]
+    safeCmd $ app "define" [Builder.fromText v <> "::"
+                             <> unType (fnType ((\(_,tp) -> viewSome yicesType tp) <$> args) (yicesType rtp))
+                           , renderTerm (yicesLambda args t)
+                           ]
 
   declareStructDatatype _ _ = return ()
 
-  writeCommand conn (Cmd cmd) =
+  writeCommand conn (YicesCommand earlyUnsatSafe cmd) =
```

Comment:
We didn't actually add the code here to skip sending the unsafe commands... oops

It happens to work, I guess, because we just skip the ACK phase for those commands, but we should avoid sending the commands also.

Reply:
Oops! Good catch!

Reply:
Fixed in c620a85c2a.